internal/support/notifications/sending_service.go

```diff
@@ -141,3 +135,101 @@ func handleFailedTransmission(t models.Transmission) {
 		}
 	}
 }
+
+func deduceAuth(s SmtpInfo) (mail.Auth, error) {
+	if s.Username == "" && s.Password == "" {
+		return nil, nil
+	}
+	if s.Username != "" && s.Password == "" {
+		return nil, errors.New("Notifications: Expecting password")
+	}
+	if s.Username == "" && s.Password != "" {
+		return nil, errors.New("Notifications: Expecting username")
+	}
+	return mail.PlainAuth("", s.Username, s.Password, s.Host), nil
+}
+
+// The function smtpSend replicates the functionality provided by the SendMail function
+// from smtp package. A rivision of standard function was needed because smtp.SendMail
+// does not allow for set-reset of InsecureSkipVerify flag of tls.Config structure. This
+// flag is needed to be manipulated for allowing the self-signed certificates.
+//
+// As it is replicating the functionality from smtp.SendMail, it borrows heavily from the
+// original function in its design and implementation. This version adds new functionality
+// for handling the SmtpInfo configuration and authentication management, along with the
+// requirement of ability to set-reset the InsecureSkipVerify flag.
+//
+// This is using a lot of unexported methods and types from smtp package through exported
+// interfaces, which makes it a little bit trickier to modify. Since, the intention for
+// this function is to use it as a support function for handling the low level SMTP
+// protocol mechanism, it is not exported.
+//
+// The addresses in the to parameter are the SMTP RCPT addresses.
+//
+// The msg parameter should be an RFC 822-style email with headers
+// first, a blank line, and then the message body. The lines of msg
+// should be CRLF terminated. The msg headers should usually include
+// fields such as "From", "To", "Subject", and "Cc".  Sending "Bcc"
+// messages is accomplished by including an email address in the to
+// parameter but not including it in the msg headers.
+//
+// The smtpSend provide no support for DKIM signing, MIME
+// attachments (see the mime/multipart package), or other mail
+// functionality.
+
```

Comment:
Can we remove this extra blank line?

Reply:
Removed part of comment also, thought that is more suitable for an exported function.